Source/WebCore/html/DOMFormData.cpp

```diff
@@ -43,13 +44,24 @@ DOMFormData::DOMFormData(ScriptExecutionContext* context, const PAL::TextEncodin
 {
 }
 
-ExceptionOr<Ref<DOMFormData>> DOMFormData::create(ScriptExecutionContext& context, HTMLFormElement* form)
+ExceptionOr<Ref<DOMFormData>> DOMFormData::create(ScriptExecutionContext& context, HTMLFormElement* form, HTMLElement* submitter)
 {
+    // https://xhr.spec.whatwg.org/#dom-formdata
     auto formData = adoptRef(*new DOMFormData(&context));
     if (!form)
         return formData;
-    
-    auto result = form->constructEntryList(nullptr, WTFMove(formData), nullptr);
+
+    RefPtr<HTMLFormControlElement> control;
+    if (submitter) {
+        if (!is<HTMLFormControlElement>(submitter))
```

Comment:
Would look nicer like so:
```
if (submitter) {
    control = dynamicDowncast<HTMLFormControlElement>(*submitter);
    if (!control || control->isSubmitButton())
        return Exception { TypeError, "The specified element is not a submit button."_s };
    if (control->form() != form)
        return Exception { NotFoundError, "The specified element is not owned by this form element."_s };
}
```

Reply:
I fixed my code sample, it used to say `control->isSubmitButton()` instead of `!control->isSubmitButton()`.

Reply:
Haha, good catch, I didn't even notice that, but my tests just caught it as well 😅